src/JobScheduler.ts

```diff
@@ -113,6 +143,32 @@ export class JobScheduler implements JobApplication {
         }
     }
 
+    private async onCommandMessage(msg: CommandMessage): Promise<boolean> {
+        let job: Job;
+        const jobRepo = this._databaseService.getJobRepository();
+        switch (msg.command) {
+            case CMD_CANCEL:
+                job = await jobRepo.findOne({ id: msg.jobId });
+                if (job && job.status === JobStatus.Queueing || job.status === JobStatus.Pause) {
+                    job.status = JobStatus.Canceled;
+                    await jobRepo.save(job);
+                    return true;
+                }
+                break;
+            case CMD_RESUME:
+                job = await jobRepo.findOne({ id: msg.jobId });
+                if (job && job.status === JobStatus.Pause) {
+                    job.status = JobStatus.Queueing;
+                    await jobRepo.save(job);
+                    await this._rabbitmqService.publish(JOB_EXCHANGE, '', job.jobMessage);
+                    return true;
+                }
+                break;
+            // no default
```

Comment:
why?

Reply:
JobExecutor and JobScheduler both subscribe to the COMMAND_QUEUE, as the queue consumers use the prefetch feature, the message will be removed from the queue, if this is not a command that JobScheduler interests in, it should NACK the message (here return false) and the message will be put back at the head of the queue for other consumers to use.

There is one JobScheduler and one or more JobExecutor running at any time. The JobScheduler only cares about some special status of a job, which will not involve a JobManager (not running).

Reply:
I think we should at least comment it here.